Forms/frmMain.cs

```diff
@@ -345,32 +428,36 @@ private void autoUpdateTimer_Tick(object sender, EventArgs e)
         /// <param name="e"></param>
         private void logUpdateTimer_Tick(object sender, EventArgs e)
         {
-            foreach(Logger.Entry logItem in Logger.items)
+            try
             {
-                ListViewItem row = new ListViewItem();
-
-                switch (logItem.Level)
+                foreach (Logger.Entry logItem in Logger.items)
                 {
-                    case Logger.Level.Warning:
-                        row.ImageIndex = 1;
-                        break;
-
-                    case Logger.Level.Error:
-                        row.ImageIndex = 2;
-                        break;
-
-                    default: //Logger.Level.Info
-                        row.ImageIndex = 0;
-                        break;
+                    ListViewItem row = new ListViewItem();
+
+                    switch (logItem.Level)
+                    {
+                        case Logger.Level.Warning:
+                            row.ImageIndex = 1;
+                            break;
+
+                        case Logger.Level.Error:
+                            row.ImageIndex = 2;
+                            break;
+
+                        default: //Logger.Level.Info
+                            row.ImageIndex = 0;
+                            break;
+                    }
+                    row.SubItems.Add(logItem.Message);
+                    this.Invoke(new addLogEntryInvoker(addLogEntry), row);
                 }
-                row.SubItems.Add(logItem.Message);
-                this.Invoke(new addLogEntryInvoker(addLogEntry), row);
+                Logger.reset(); //Clear the log
+            }
+            catch (Exception)

```

Comment:
What exceptions are being caught here? Would be better to handle them properly?

Reply:
very rare Exception had it once so i take this solution, because i cant reproduce it.